### PR TITLE
Add new question type table supporting nested tables

### PR DESCRIPTION
**Pull request by sliverc**
* Introduce new answer type AnswerTable
* A table  format is represented by a form
* A single row is represented as a document of question row form
* JEXL expanded whereas a table answer is a list of dicts with question
  keys.
* Introduce document family to allow access to a tree of documents
  introduced through table answer.

### CODE REVIEW

**Comment by sliverc** [comment]
Sorry for the lengthy PR but is a bit of an advanced feature.

File: caluma/form/migrations/0003_auto_20190128_0836.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                ),
+            ],
+            options={"ordering": ("-sort", "id")},
+        ),
+        migrations.AddField(
+            model_name="question",
+            name="row_form",
+            field=models.ForeignKey(
+                blank=True,
+                help_text="One row of table is represented by this form",
```
Comment (czosel) [inline caluma/form/migrations/0003_auto_20190128_0836.py]: Some ideas:
`One row of a table is represented by this form`
`Form that represents a row of the table`
`Form used to enter each row of the table`

Comment (sliverc) [reply caluma/form/migrations/0003_auto_20190128_0836.py]: used second suggestion

**Comment by sliverc** [comment]
Rebased to master and also made sure that only one new migrations is created.

File: caluma/form/serializers.py [hunk 2]
```diff
@@ -324,3 +333,82 @@ class SaveDocumentFloatAnswerSerializer(SaveAnswerSerializer):
...
         pass
+
+
+class SaveDocumentTableAnswerSerializer(SaveAnswerSerializer):
+    value = serializers.GlobalIDPrimaryKeyRelatedField(
+        source="documents",
+        queryset=models.Document.objects,
+        many=True,
+        required=True,
+        help_text="List of document ids whereas one document is one row in the table",
```
Comment (czosel) [inline caluma/form/serializers.py]:
```suggestion
        help_text="List of document IDs, each representing one row in the table",
```

Comment (czosel) [reply caluma/form/serializers.py]: or probably even better: "List of document IDs representing the rows in the table"

File: caluma/form/serializers.py [hunk 3]
```diff
@@ -324,3 +333,82 @@ class SaveDocumentFloatAnswerSerializer(SaveAnswerSerializer):
...
+        for child_document in child_documents:
+            yield from self._get_document_tree(child_document.pk)
+
+        yield document_id
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        documents = validated_data.pop("documents")
+
+        # detach answers to its own family tree
```
Comment (czosel) [inline caluma/form/serializers.py]: The comment is confusing me a bit. Aren't we detaching answers from their old family tree and afterwards reattaching them to the new one?

File: caluma/form/serializers.py (hunk 2 above)
Comment (sliverc) [reply caluma/form/serializers.py]: Adjusted

**Pull request merged**